src/main/java/organice/model/ModelManager.java

```diff
@@ -152,7 +157,39 @@ public boolean equals(Object obj) {
         ModelManager other = (ModelManager) obj;
         return addressBook.equals(other.addressBook)
                 && userPrefs.equals(other.userPrefs)
-                && filteredPersons.equals(other.filteredPersons);
+                && filteredPersons.equals(other.filteredPersons)
+                && sortedPersons.equals(other.sortedPersons);
     }
 
 }
+
+//=========== Sorted Person List Accessors =============================================================
+
+    /**
+     * Returns a sorted view of the list of {@code Person} backed by the internal list of
+     * {@code versionedAddressBook}
+     */
+    @Override
+    public SortedList<Person> getSortedPersonList() {
+        return sortedPersons;
+    }
+
+    @Override
+    public void sortByPriority() {
+        sortedPersons = new SortedList<>(filteredPersons);
```

Comment:
Just to take note, my implementation stores the matches in a separate list called `listOfMatches`. Refer to my branch for more information.